crates/zed/src/languages/dockerfile.rs

```
@@ -0,0 +1,124 @@
+use anyhow::{anyhow, Result};
+use async_trait::async_trait;
+use futures::StreamExt;
+use language::{LanguageServerName, LspAdapter, LspAdapterDelegate};
+use lsp::LanguageServerBinary;
+use node_runtime::NodeRuntime;
+use smol::fs;
+use std::{
+    any::Any,
+    ffi::OsString,
+    path::{Path, PathBuf},
+    sync::Arc,
+};
+use util::ResultExt;
+
+const SERVER_PATH: &'static str =
+    "node_modules/dockerfile-language-server-nodejs/bin/docker-langserver";
+
+fn server_binary_arguments(server_path: &Path) -> Vec<OsString> {
+    vec![server_path.into(), "--stdio".into()]
+}
+
+pub struct DockerfileLspAdapter {
+    node: Arc<dyn NodeRuntime>,
+}
+
+impl DockerfileLspAdapter {
+    pub fn new(node: Arc<dyn NodeRuntime>) -> Self {
+        Self { node }
+    }
+}
+
+#[async_trait]
+impl LspAdapter for DockerfileLspAdapter {
+    fn name(&self) -> LanguageServerName {
+        LanguageServerName("dockerfile".into())
```

For the full language server name I think we'll want to use the name of the "binary":

```suggestion
        LanguageServerName("docker-langserver".into())
```